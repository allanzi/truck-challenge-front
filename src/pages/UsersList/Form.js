/* eslint-disable no-undef */

import React, { useState, useEffect } from 'react';
import {
  Modal, Form, Input, Spin, Row, Col, Switch, InputNumber, message
} from 'antd';

import truckApi from '../../services/api.service';

const UserFormPage = ({ params, isUpdate }) => {
  const [loading, setLoading] = useState(false);
  const [form] = Form.useForm();

  // eslint-disable-next-line consistent-return
  const getUser = async () => {
    try {
      setLoading(true);
      const { data: { data } } = await truckApi.showUser(params.id);
      form.setFieldsValue(data);
    } catch (e) {
      if (isUpdate) {
        return message.error('Não foi possível editar esse caminhoneiro! Tente novamente mais tarde.');
      }

      message.error('Não foi possível cadastrar esse caminhoneiro! Tente novamente mais tarde.');
    } finally {
      setLoading(false);
    }
  };

  const handleRequest = async () => {
    try {
      let data = await form.validateFields();
      data = {
        ...data,
        has_vehicle: data.has_vehicle || false,
        is_busy: data.is_busy || false,
      };

      if (isUpdate) {
        await truckApi.updateUser(params.id, data);
        message.success('Caminhoneiro atualizado com sucesso');
      } else {
        await truckApi.createUser(data);
        message.success('Caminhoneiro criado com sucesso');
      }

      // eslint-disable-next-line no-return-assign
      setTimeout(() => window.location.pathname = '/users', 1500);
    } catch (e) {
      message.error('Corrija os campos!');
    }
  };

  const handleCancel = () => {
    window.location.pathname = '/users';
  };

  useEffect(() => {
    if (isUpdate) {
      getUser();
    }

    document.title = 'Truck Panel | Caminhoneiros';
  }, []);

  return (
    <Modal
      visible
      title={isUpdate ? 'Editar Caminhoneiro' : 'Cadastrar Caminhoneiro'}
      okText={isUpdate ? 'Atualizar' : 'Cadastrar'}
      cancelText="Cancelar"
      onOk={handleRequest}
      onCancel={handleCancel}
    >
      {loading ? (
        <Row justify="center">
          <Col>
            <br />
            <br />
            <br />
            <Spin />
            <br />
            <br />
            <br />
          </Col>
        </Row>
      ) : (
        <>
          <Form
            form={form}
            layout="vertical"
            name="edit_travel"
          >
            <Row gutter={[8, 8]}>
              <Col span={8}>
                <Form.Item
                  rules={[
                    { required: true, message: 'Este campo é obrigatório!' },
                    { max: 60, message: 'Limite de caracteres 60!' },
                  ]}
                  name="name"
                  label="Nome"
                >
                  <Input />
                </Form.Item>
              </Col>
              <Col span={8}>
                <Form.Item
                  rules={[
                    { required: true, message: 'Este campo é obrigatório!' },
                  ]}
                  name="age"
                  label="Idade"
                >
                  <InputNumber
                    min={18}
                    max={100}
                    style={{
                      width: '100%',
                    }}
                  />
                </Form.Item>
              </Col>
              <Col span={8}>
                <Form.Item
                  rules={[
                    { required: true, message: 'Este campo é obrigatório!' },
                    { max: 5, message: 'Limite de caracteres 5!' }
                  ]}
                  name="driver_license_type"
                  label="Tipo CNH?"
                >
                  <Input />
                </Form.Item>
              </Col>
            </Row>
            <Row gutter={[8, 8]}>
              <Col span={8}>
                <Form.Item
                  name="has_vehicle"
                  label="Possui veículo?"
                >
                  <Switch />
                </Form.Item>
              </Col>
              <Col span={8}>
                <Form.Item
                  name="is_busy"
                  label="Está carregado?"
                >
                  <Switch />
                </Form.Item>
              </Col>
              <Col span={8}>
                <Form.Item
                  rules={[{ required: true, message: 'Este campo é obrigatório!' }]}
                  name="vehicle_type_id"
                  label="Tipo veículo"
                >
                  <InputNumber
                    style={{
                      width: '100%',
                    }}
                  />
                </Form.Item>
              </Col>
            </Row>
          </Form>
        </>
      )}
    </Modal>
  );
};

export default UserFormPage;
