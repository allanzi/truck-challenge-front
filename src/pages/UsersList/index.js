import React, { useEffect, useState } from 'react';
import {
  Breadcrumb, Table, Popconfirm, message, Button, Row, Col
} from 'antd';

import truckApi from '../../services/api.service';
import './styles.css';
import TravelForm from './Form';

const UsersListPage = ({ match: { params } }) => {
  const [users, setUsers] = useState({ loading: true, data: [] });
  const [visibleForm, setVisibleForm] = useState(false);
  const [isUpdate, setIsUpdate] = useState(true);

  const getUsers = async () => {
    const { data: { data } } = await truckApi.listUsers();

    setTimeout(() => setUsers({ loading: false, data }), 500);
  };

  const handleDelete = async (id) => {
    try {
      await truckApi.deleteUser(id);
      message.success('Excluído com sucesso');
      getUsers();
    } catch {
      message.error('Não foi possível excluir, tente novamente mais tarde.');
    }
  };

  const handleEditButton = (id) => {
    window.location.pathname = `/users/${id}`;
  };

  const handleCreateButton = () => {
    window.location.pathname = '/users/create';
  };

  useEffect(() => {
    getUsers();
    document.title = 'Truck Panel | Caminhoneiros';

    if ('id' in params && params.id) {
      if (params.id === 'create') {
        setIsUpdate(false);
      }

      setVisibleForm(true);
    }
  }, []);

  return (
    <>
      <Row justify="space-between" align="middle">
        <Col span={12}>
          <Breadcrumb style={{ margin: '16px 0' }}>
            <Breadcrumb.Item>Truck Panel</Breadcrumb.Item>
            <Breadcrumb.Item>Caminhoneiros</Breadcrumb.Item>
          </Breadcrumb>
        </Col>
        <Col spam={12}>
          <Button type="primary" onClick={handleCreateButton}>Cadastrar caminhoneiro</Button>
        </Col>
      </Row>
      <div className="site-layout-background" style={{ padding: 24, minHeight: 360 }}>
        <Table
          rowKey={(item) => item.name}
          loading={users.loading}
          dataSource={users.data}
          locale={{
            emptyText: 'Sem dados para exibir',
          }}
          columns={[
            {
              title: 'Nome',
              dataIndex: 'name',
              key: 'name',
            },
            {
              title: 'Idade',
              dataIndex: 'age',
              key: 'age',
            },
            {
              title: 'Tipo CNH?',
              dataIndex: 'driver_license_type',
              key: 'driver_license_type',
              render: (item) => (<span>{item ? 'Sim' : 'Não'}</span>)
            },
            {
              title: 'Possui veículo?',
              dataIndex: 'has_vehicle',
              key: 'has_vehicle',
              render: (item) => (<span>{item ? 'Sim' : 'Não'}</span>)
            },
            {
              title: 'Está carregado?',
              dataIndex: 'is_busy',
              key: 'is_busy',
              render: (item) => (<span>{item ? 'Sim' : 'Não'}</span>)
            },
            {
              title: 'Tipo veículo',
              dataIndex: 'vehicle_type_id',
              key: 'vehicle_type_id',
            },
            {
              title: 'Ações',
              key: 'action',
              render: (text, record) => (
                <span>
                  <a style={{ marginRight: 16 }} onClick={() => handleEditButton(record.id)}>Editar</a>
                  <Popconfirm
                    title="Realmente deseja excluir?"
                    onConfirm={() => handleDelete(record.id)}
                    okText="Sim"
                    cancelText="Não"
                  >
                    <a href="javascript:void(0);">Excluir</a>
                  </Popconfirm>
                </span>
              ),
            },
          ]}
        />
        {visibleForm && (
          <TravelForm
            params={params}
            isUpdate={isUpdate}
          />
        )}
      </div>
    </>
  );
};

export default UsersListPage;
