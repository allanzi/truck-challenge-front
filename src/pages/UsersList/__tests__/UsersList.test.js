import React from 'react';

import UsersList from '../index';

it('should render correct', () => {
  const wrapper = shallow(<UsersList match={{ params: { id: 'create' } }} />);

  expect(wrapper).toMatchSnapshot();
});
