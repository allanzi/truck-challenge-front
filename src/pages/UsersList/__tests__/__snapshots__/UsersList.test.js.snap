// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct 1`] = `
<Fragment>
  <Row
    align="middle"
    gutter={0}
    justify="space-between"
  >
    <Col
      span={12}
    >
      <Breadcrumb
        separator="/"
        style={
          Object {
            "margin": "16px 0",
          }
        }
      >
        <BreadcrumbItem
          separator="/"
        >
          Truck Panel
        </BreadcrumbItem>
        <BreadcrumbItem
          separator="/"
        >
          Caminhoneiros
        </BreadcrumbItem>
      </Breadcrumb>
    </Col>
    <Col
      spam={12}
    >
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        Cadastrar caminhoneiro
      </Button>
    </Col>
  </Row>
  <div
    className="site-layout-background"
    style={
      Object {
        "minHeight": 360,
        "padding": 24,
      }
    }
  >
    <Table
      columns={
        Array [
          Object {
            "dataIndex": "name",
            "key": "name",
            "title": "Nome",
          },
          Object {
            "dataIndex": "age",
            "key": "age",
            "title": "Idade",
          },
          Object {
            "dataIndex": "driver_license_type",
            "key": "driver_license_type",
            "render": [Function],
            "title": "Tipo CNH?",
          },
          Object {
            "dataIndex": "has_vehicle",
            "key": "has_vehicle",
            "render": [Function],
            "title": "Possui veículo?",
          },
          Object {
            "dataIndex": "is_busy",
            "key": "is_busy",
            "render": [Function],
            "title": "Está carregado?",
          },
          Object {
            "dataIndex": "vehicle_type_id",
            "key": "vehicle_type_id",
            "title": "Tipo veículo",
          },
          Object {
            "key": "action",
            "render": [Function],
            "title": "Ações",
          },
        ]
      }
      dataSource={Array []}
      loading={true}
      locale={
        Object {
          "emptyText": "Sem dados para exibir",
        }
      }
      rowKey={[Function]}
    />
  </div>
</Fragment>
`;
